### ISSUE TITLE
Allow a screenshot to be taken from the General tab

### ISSUE THREAD

**Issue by AnthonyTacquet** [Ylianst/MeshCentral#2289]
Is it possible to add an extra button next to Chat/Meet/Share to allow a screenshot to be taken from the Desktop (it is less intrusive for the user and sometimes it is already enough to troubleshoot).

**Comment by Ylianst** [comment]
Not sure what "less intrusive to the user" would be, it would still require asking for user consent if that feature is enabled. I will chat with Bryan about this, we may still need to spawn a different process to get the screen, so internally, it's almost the same at remote desktop.

**Comment by PathfinderNetworks** [comment]
When I just need to see the desktop of a remote device and don't want to control the input (such as moving their mouse) I just deselect the Input radio button at the bottom of the window before I connect.  Doing that means I can view the remote desktop without concern for anything I do on my end interrupting what the remote user is doing.  Hope that makes sense.

**Comment by jensborgenon** [comment]
a screenshot would be a very nice addition, it would allow you to see the history of an issue when you have a timed screenshot stored, if you have to take over the screen and make a printscreen that would take a lot more clicks

**Comment by DMoRiaM** [comment]
Hi!

I agree that the screenshot could show itself be useful. As exemple, one 
user of mine, with access to his computers, asked me how to take screen 
shots from one of his employees so it could be used in a meeting to 
inform the user about an unwanted behavior (using NSFW sites during 
working hours). I teached him to use the Windows Snipping Tool. And he 
needed a lot of extra steps to make it work well (maximize screen, call 
the tool, select te screen, save the file, etc). As a lay user and a old 
aged man, some times he strugles with that.

In trail of the sugestion I believe what could be a good idea where to 
allow the user also take a screenshot every time he or she found some 
error condition, like a popup message, and that screenshot be listed 
somewhere with a checkbox (for "us" to mark what we already took care) 
and a delete button (for when it is no more needed).

Regards

Em 12/02/2021 10:32, Anthony Tacquet escreveu:

**Comment by Ylianst** [comment]
I will chat with Bryan on this. As long as the user consent policies allow it, I would try to work on this.

**Comment by matthiscock** [comment]
I would def really like to see this feature also.
I saves having to remote in, or just know a computer is being used.....

**Comment by openncomp** [comment]
I agree, a one-button screenshot feature in MeshCentral would be a great idea!

**Comment by matthiscock** [comment]
Any news on this? @Ylianst

**Comment by krayon007** [comment]
Since I'm writing an updated KVM module for windows, I can add a screenshot function to it at the same time.

**Comment by samschultzponsys** [comment]
Per @si458 in the meeting I will reopen this as this is still something I think the community is after.

**Comment by VitoFe** [comment]
Following up on the discussion around privacy concerns and user consent when triggering screenshots from the admin/server side, I wanted to propose a user-initiated alternative (or addition). Instead of (or even better in addition to) an admin-triggered screenshot, we could introduce a user-triggered reporting feature from the agent side (via the MeshCentral Assistant UI or a taskbar icon menu):

- User-Initiated Screenshot: A button in the agent UI or taskbar menu that the user can click to capture their current screen and send it directly to the MeshCentral server, making it visible to the supporting administrator.
- Rolling "Last N Seconds" Buffer: To make this even more powerful, the local agent could maintain a lightweight, rolling buffer of the last N seconds of the screen feed (captured locally at a low framerate/quality and stored strictly temporarily on the client). If the user encounters an error, they click the button, and this short recording is sent to the server.

Troubleshooting errors that are difficult to replicate is a common pain point, by the time an administrator connects, the error state is gone. Having a short video buffer of the seconds leading up to the user clicking the button would show exactly what happened, making it much easier to diagnose.

I would love to hear thoughts on this approach and whether implementing something like this in the agent or MeshCentral Assistant is feasible.

**Comment by samschultzponsys** [comment]
@VitoFe This is good but we already have software that handles screenshots I just want to retire the other software as its super slow old and runs directly ONLY on Windows. I do not mind if the screen shows that a screenshot is being taken thats fine with me. Support has needs for screenshots that are not user initiated.
